### ISSUE TITLE
Why do we train on Bounding box ids, instead of Track_ids?

### ISSUE THREAD

**Issue by oljike** [ifzhang/FairMOT#312]
Hi! I have a question about the linear layer which is created by 
`self.classifier = nn.Linear(self.emb_dim, self.nID)`
The `self.nID` parameter is the total number of our bounding boxes, not the total number of Track_ids. 
Also, in the loss function the target is defined as follows:
`id_target = batch['ids'][batch['reg_mask'] > 0]`
where batch['ids'] is also the bounding box id, not the track_id. 
So, the question is: why we treat each individual bounding box as a separate class, instead of using track_ids as our target classes? Thanks.

**Comment by ifzhang** [comment]
For CrowdHuman dataset, we use bbox id because it doesn't have track id. For other MOT dataset, we use track id.
